test_single_controller_ppo.py

```diff
@@ -697,6 +742,12 @@ def get_next_iter_prompts(self):
 
         return preprocess_batches(batches, self.generations_per_prompt, self.tokenizer.pad_token_id)
 
+    def get_dataloader_state_dict(self):
+        return self.dataloader.state_dict()
+    
+    def load_dataloader_state_dict(self, state_dict: dict):
+        self.dataloader.load_state_dict(state_dict)
```

Comment:
```suggestion
        self.dataloader.load_state_dict(state_dict)
        self.dataloader_iter= iter(self.dataloader)
```
need to check if this can resume from where it breaks or start